src/views/Generator/TestData/DataFiles.tsx

```diff
@@ -0,0 +1,129 @@
+  const setFiles = useGeneratorStore((store) => store.setFiles)
+  const options = [...availableFiles.values()].filter(
+    (file) => !selectedFiles.find((f) => f.name === file.fileName)
+  )
+
+  const handleAdd = (fileName: string) => {
+    setFiles([...selectedFiles, { name: fileName }])
+  }
+
+  const handleRemove = (fileName: string) => {
+    setFiles(selectedFiles.filter((file) => file.name !== fileName))
+  }
+
+  return (
+    <>
+      <Text size="2" as="p" mb="2">
+        Configure data files and use them in you test rules.
+      </Text>
+      <Table.Root size="1" variant="surface">
+        <Table.Header>
+          <Table.Row>
+            <Table.ColumnHeaderCell>Data file</Table.ColumnHeaderCell>
+            <Table.ColumnHeaderCell>
+              <Flex align="center" gap="1">
+                Access method
+                <Tooltip content="Defines how the items are accessed during the test">
+                  <InfoCircledIcon />
+                </Tooltip>
+              </Flex>
+            </Table.ColumnHeaderCell>
+            <Table.ColumnHeaderCell width="0"></Table.ColumnHeaderCell>
+          </Table.Row>
+        </Table.Header>
+
+        <Table.Body>
+          {selectedFiles.map((file) => (
+            <DataFileRow
+              key={file.name}
+              file={file}
+              onRemove={() => handleRemove(file.name)}
+            />
+          ))}
+          {options.length > 0 && (
+            <Table.Row>
+              <Table.RowHeaderCell colSpan={3} justify="center">
+                <DropdownMenu.Root>
+                  <DropdownMenu.Trigger>
+                    <Button variant="ghost">
+                      Add data file <PlusIcon />
+                    </Button>
+                  </DropdownMenu.Trigger>
+                  <DropdownMenu.Content>
+                    {options.map((file) => (
+                      <DropdownMenu.Item
+                        key={file.fileName}
+                        onClick={() => handleAdd(file.fileName)}
+                      >
+                        {file.fileName}
+                      </DropdownMenu.Item>
+                    ))}
+                  </DropdownMenu.Content>
+                </DropdownMenu.Root>
+              </Table.RowHeaderCell>
+            </Table.Row>
+          )}
+        </Table.Body>
+      </Table.Root>
+    </>
+  )
+}
+
+interface DataFileRowProps {
+  file: DataFile
+  onRemove: () => void
+}
+
+function DataFileRow({ file, onRemove }: DataFileRowProps) {
+  const isFileInUse = useGeneratorStore((state) =>
+    state.rules.some(
+      (rule) =>
+        rule.type === 'parameterization' &&
+        rule.value.type === 'dataFileValue' &&
+        rule.value.fileName === file.name
+    )
+  )
+
+  return (
+    <Table.Row
+      css={css`
+        vertical-align: middle;
+      `}
+    >
+      <Table.Cell>{file.name}</Table.Cell>
+      <Table.Cell>Unique item per iteration</Table.Cell>
```

Comment:
Maybe `Random item per iteration` is a bit more clear?

Reply:
It's not random 😬 it's access array values based on `scenario.iterationInTest` as described here. I'll try to come up with better copy